src/main/java/com/novi/backend_spring_boot_tech_it_easy_controller/services/WallBracketService.java

```diff
@@ -0,0 +1,78 @@
+package com.novi.backend_spring_boot_tech_it_easy_controller.services;
+
+import com.novi.backend_spring_boot_tech_it_easy_controller.dtos.WallBracketDto;
+import com.novi.backend_spring_boot_tech_it_easy_controller.dtos.WallBracketInputDto;
+import com.novi.backend_spring_boot_tech_it_easy_controller.models.WallBracket;
+import com.novi.backend_spring_boot_tech_it_easy_controller.repositories.WallBracketRepository;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+
+import java.util.ArrayList;
+import java.util.List;
+
+@Service
+public class WallBracketService {
+
+    private final WallBracketRepository wallBracketRepository;
+
+    @Autowired
+    public WallBracketService(WallBracketRepository wallBracketRepository) {
+        this.wallBracketRepository = wallBracketRepository;
+    }
+
+    public List<WallBracketDto> getAllWallBrackets() {
+        List<WallBracket> wallBracketsList = wallBracketRepository.findAll();
+        List<WallBracketDto> wallBracketDtoList = new ArrayList<>();
+
+        for (WallBracket wallBracket : wallBracketsList) {
+            WallBracketDto wallBracketDto = convertToDto(wallBracket);
+            wallBracketDtoList.add(wallBracketDto);
+        }
+
+        return wallBracketDtoList;
+    }
+
+    public WallBracketDto getWallBracketById(Long id) {
+        WallBracket wallBracket = wallBracketRepository.findById(id)
+                .orElseThrow(() -> new RuntimeException("WallBracket not found"));
+        return convertToDto(wallBracket);
+    }
+
+    public WallBracketDto createWallBracket(WallBracketInputDto wallBracketInputDto) {
+        WallBracket wallBracket = new WallBracket();
+        wallBracket.setName(wallBracketInputDto.getName());
+        wallBracket.setBrand(wallBracketInputDto.getBrand());
+        wallBracket.setPrice(wallBracketInputDto.getPrice());
+
+        wallBracket = wallBracketRepository.save(wallBracket);
+        return convertToDto(wallBracket);
+    }
+
+    public WallBracketDto updateWallBracket(Long id, WallBracketDto wallBracketInputDto) {
+        WallBracket wallBracket = wallBracketRepository.findById(id)
+                .orElseThrow(() -> new RuntimeException("WallBracket not found"));
```

Comment:
Hier dus ook de verkeerde exception

Reply:
Het is niet de bedoeling dat je kale RuntimeExceptions gaat gooien. Zo kun je in je log (je console in ons geval) niet aflezen welke error er is gegooid en waar dat probleem kan liggen